src/test/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinitionTest.java

```diff
@@ -195,6 +201,32 @@ public class CpsScmFlowDefinitionTest {
         r.assertLogContains("version one", b);
     }
 
+    @Test public void lightweight_brach_parametrised() throws Exception {
+        LoggerRule lr = new LoggerRule();
+        lr.record(GitSCMFileSystem.class.getName(), Level.ALL).capture(4024);
+
+        sampleRepo.init();
+        sampleRepo.git("checkout","-b","master2");
+        sampleRepo.write("flow.groovy", "echo 'version one'");
+        sampleRepo.git("add", "flow.groovy");
+        sampleRepo.git("commit", "--message=init");
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.addProperty(new ParametersDefinitionProperty(new StringParameterDefinition("BRANCH","","")));
+        GitStep step = new GitStep(sampleRepo.toString());
+        step.setBranch("${BRANCH}");
+
+        CpsScmFlowDefinition def = new CpsScmFlowDefinition(step.createSCM(), "flow.groovy");
+        def.setLightweight(true);
+        TestDurabilityHintProvider provider = Jenkins.get().getExtensionList(TestDurabilityHintProvider.class).get(0);
+        provider.registerHint("p", FlowDurabilityHint.PERFORMANCE_OPTIMIZED);
+        p.setDefinition(def);
+
+        WorkflowRun b = r.assertBuildStatusSuccess(p.scheduleBuild2(0, new ParametersAction(new StringParameterValue("BRANCH","master2"))));
+        Assert.assertEquals(FlowDurabilityHint.PERFORMANCE_OPTIMIZED, b.getExecution().getDurabilityHint());
+
+        assertThat(lr.getMessages(), hasItem(containsString("refs/heads/master2:refs/remotes/origin/master2")));
```

Comment:
Probably a more straightforward way to assert this. Or just delete entirely—the fact that the build ran at all (when the repo only contains the `master2` branch) is proof enough that the fix worked, right?

Reply:
I wasn't sure here, because I saw other tests parsing the log file, so I thought that this would be a proof. Also I am not sure if the sample repo doesn't create master automatically now or in the future

Reply:
Oki I simplified the test, and somehow created the scm directly, removed the useless log recording check, and the performence hits.